app/commands/concerns/log_entry_command.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module LogEntryCommand
+  extend ActiveSupport::Concern
+
+  def build_log_entry(type)
+    @log_entry = LogEntry.new(
+      signed_data: signed_data,
+      log_type: type,
+      client: client
+    )
+  end
+
+  def log_entry_validations
+    if decoded_data.blank?
+      "Invalid signature"
```

Comment:
Oh, one more comment. 😄 Will we use Internationalization? If so, we might want to add it before we end up with lots of locale strings.

Reply:
yes, we will address that in the future, probably adding an error code to the message.